build/circle-deps.sh

```diff
@@ -85,3 +107,48 @@ grep -v '^cmd' GLOCKFILE | glock sync -n
 # Recursively invoke this script inside the builder docker container,
 # passing "docker" as the first argument.
 $(dirname $0)/builder.sh $0 docker
+
+# According to
+# https://discuss.circleci.com/t/cache-save-restore-algorithm/759, the
+# cache is only collected from container (shard) 0. We work around
+# this limitation by copying the build output from shards 1 and 2 over
+# to shard 0. On shard 0 we wait for this remote data to be copied.
+
+gopath0="${GOPATH%%:*}"
+
+if is_shard 2; then
+  # We might already be on shard 0 if we're running without
+  # parallelism. Avoid deleting our build output in that case.
+  if ! is_shard 0; then
+    dir="${HOME}/uicache"
+    time rsync -av --delete "${dir}/" node0:"${dir}"
+
+    cmds=$(grep '^cmd' GLOCKFILE | grep -v glock | awk '{print $2}' | awk -F/ '{print $NF}')
+    time ssh node0 mkdir -p "${gopath0}/bin/linux_amd64"
+    for cmd in ${cmds}; do
+      path="${gopath0}/bin/linux_amd64/${cmd}"
+      time rsync -v "${path}" node0:"${path}"
```

Comment:
Hm. I thought it would still be necessary to preserve all the file's attributes (like executability).